### DISCUSSION TITLE
null-ls integration

### DISCUSSION THREAD

**Discussion by axieax** [mason-org/mason.nvim#143] [Q&A] [answered]
Hi, just wondering how to integrate mason-installed binaries with null-ls. Can't find any clear instructions on how to do so.

**Comment by axieax** [comment, marked as answer]
Update: from my understanding these binaries just get added to the vim PATH

**Reply by williamboman** [reply]
Hello! Yes that is correct. I will probably add an FAQ section to thw readme explaining this. Mason is not involved in integrating these tools in Neovim - that's provided by builtin mechanisms as well as other plugins

**Reply by axieax** [reply]
Yep that would be very helpful! May I also ask what the point of https://github.com/WhoIsSethDaniel/mason-tool-installer.nvim is? Seems like I can just pass my desired packages to mason_lspconfig's ensure_installed, but that doesn't work for some reason.

**Reply by williamboman** [reply]
The `ensure_installed` setting in `mason-lspconfig` only applies to LSP servers (it also only accepts lspconfig server names). It's a remnant from nvim-lsp-installer that remained to make migration easier.

That plugin you linked to makes it possible to specify a list of Mason packages that you want to ensure are installed. See https://github.com/williamboman/mason.nvim/issues/103#issuecomment-1190641771 for more info why this is not provided in core yet

**Reply by axieax** [reply]
Ahh I see thanks for the explanation! One more question: is there a reason why the mason language server name does not match the lspconfig name, e.g. `sumneko_lua` vs `lua-language-server`?

**Reply by jaworek** [reply]
You can find explanation for this change in the last paragraph of this discussion thread:
https://github.com/williamboman/nvim-lsp-installer/discussions/876
